File: app/src/main/java/com/sysu/edu/api/CourseSelectionViewModel.java
package com.sysu.edu.api;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.HashMap;

public class CourseSelectionViewModel extends ViewModel {
    public final MutableLiveData<HashMap<String, String>> filterName = new MutableLiveData<>(new HashMap<>());
    public final MutableLiveData<HashMap<String, String>> filterValue = new MutableLiveData<>(new HashMap<>());
    String returnData;

    public HashMap<String, String> getFilterName() {
        return filterName.getValue();
    }

    public void setFilterName(HashMap<String, String> filter) {
        filterName.postValue(filter);
    }

    public HashMap<String, String> getFilterValue() {
        return filterValue.getValue();
    }

    public void setFilterValue(HashMap<String, String> filter) {
        filterValue.postValue(filter);
    }

    public String getReturnData() {
        return returnData == null ? "" : returnData;
    }

    public void setReturnData(String data) {
        returnData = data;
    }

    //public void clearReturnData() {
//        returnData = null;
//    }

}

File: app/src/main/java/com/sysu/edu/academic/CourseSelectionViewModel.kt
package com.sysu.edu.academic

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class CourseSelectionViewModel : ViewModel() {
    //@JvmField val filterName: MutableLiveData<MutableMap<String?, String?>?> = MutableLiveData<MutableMap<String?, String?>?>(HashMap())

    @JvmField
    val filterValue: MutableLiveData<MutableMap<String?, String?>?> =
        MutableLiveData<MutableMap<String?, String?>?>(HashMap())
    var returnData: String? = null
        get() {
            return if (field == null) "" else field
        }

    //fun getFilterName(): MutableMap<String?, String?>? {
    //	return filterName.getValue()
    //}
    //
    //fun setFilterName(filter: MutableMap<String?, String?>) {
    //	filterName.postValue(filter)
    //}
    //
    //fun getFilterValue(): MutableMap<String?, String?>? {
    //	return filterValue.getValue()
    //}
    //
    //fun setFilterValue(filter: MutableMap<String?, String?>) {
    //	filterValue.postValue(filter)
    //}
//	fun getReturnData(): String {	//		return (if (returnData == null) "" else returnData)!!
    //	}
    //
    //	fun setReturnData(data: String?) {
    //		returnData = data
    //	}
}